### PR TITLE
[WIP] Use atomic to replace lock

### PR DESCRIPTION
**Pull request by cwdsuzhou**
**What type of PR is this?**

/kind design
/sig scheduling

**What this PR does / why we need it**:

Use atomic to replace lock, can improve the performance, as https://github.com/kubernetes/kubernetes/pull/76243 mentioned

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

/assign @bsalamat
/assign @Huang-Wei 

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: staging/src/k8s.io/apimachinery/pkg/util/errors/errors.go [hunk 1]
```diff
@@ -186,8 +186,8 @@ func CreateAggregateFromMessageCountMap(m MessageCountMap) Aggregate {
 	result := make([]error, 0, len(m))
 	for errStr, count := range m {
 		var countStr string
-		if count > 1 {
-			countStr = fmt.Sprintf(" (repeated %v times)", count)
+		if *count > int32(1) {
```
Comment (misterikkit) [inline staging/src/k8s.io/apimachinery/pkg/util/errors/errors.go]: is casting the literal needed here?

File: staging/src/k8s.io/apimachinery/pkg/util/errors/errors_test.go [hunk 2]
```diff
@@ -328,24 +328,26 @@ func TestFlatten(t *testing.T) {
...
+	var errCount2 int32 =2
 	testCases := []struct {
 		name     string
 		mcm      MessageCountMap
 		expected Aggregate
 	}{
 		{
 			"input has single instance of one message",
-			MessageCountMap{"abc": 1},
+			MessageCountMap{"abc": &errCount1},
```
Comment (misterikkit) [inline staging/src/k8s.io/apimachinery/pkg/util/errors/errors_test.go]: this makes the test a bit harder to read. How about a helper function like `intPointer(int)*int`?

**Review by misterikkit — COMMENTED** [review]
Thanks for working on scheduler performance. Why did you close this PR?

Also, do you have benchmarks showing the improvement for this change? (ping me if you don't know how to run those. I think we have docs on it somewhere...)

**Comment by bsalamat** [comment]
Scheduler benchmarking instructions can be found at: https://github.com/kubernetes/community/blob/master/contributors/devel/sig-scheduling/scheduler_benchmarking.md

**Comment by cwdsuzhou** [comment]
Thanks @bsalamt @misterikkit for your advice. I tested it locally, but did not found improvement, so I closed the pr. I would run the benchmark, if I find improvement, I would reopen it

**Comment by Huang-Wei** [comment]
I think no improvement is b/c the error handling is not in the critical path - or even not hit in 99% of cases.

**Comment by cwdsuzhou** [comment]
> I think no improvement is b/c the error handling is not in the critical path - or even not hit at all in 99% of cases.

Yeah, you may be right. Anyhow, I would like to run the benchmark.

**Comment by cwdsuzhou** [comment]
@Huang-Wei @misterikkit @bsalamat 

I have run  the benchmark and I have not observe an obvious improvement.  I think that is caused by some reasons as@Huang-Wei mentioned. However, there are still some places should be changed, so I opened a new PR https://github.com/kubernetes/kubernetes/pull/76605.

**Pull request closed without merging**